File: valid-parentheses/valid-parentheses.js
/**
 * @param {string} s
 * @return {boolean}
 */
var isValid = function(s) {
    const parentheses = [];
    const open = ["(", "{", "["];
    const closed = [")", "}", "]"];
    for (let i = 0; i < s.length; i++) {
        if (open.includes(s[i])) {
            parentheses.push(s[i])
        } else if (closed.includes(s[i])) {
            const popped = parentheses.pop(); 
            if (open.indexOf(popped) !== closed.indexOf(s[i])) {
                return false;
            }
        }
    }
    return parentheses.length === 0;
};

File: valid-parentheses/valid-parentheses.py
def is_valid(s: str):
    stack = []
    open = ["(", "[", "{"]
    closed = [")", "]", "}"]
    for elt in s:
        if (elt in open):
            stack.append(elt)
        elif elt in closed:
            if len(stack) == 0:
                return False
            popped = stack.pop()
            if (closed.index(elt) != open.index(popped)):
                return False
    return len(stack) == 0

if __name__ == "__main__":
    print(is_valid("()")) # True
    print(is_valid("()[]{}")) # True
    print(is_valid("(]")) # False
    print(is_valid("({})")) # True
    print(is_valid("([)]")) # False